src/ig/IGExporter.ts

```diff
@@ -131,57 +115,68 @@ export class IGExporter {
         ]
       }
     };
-
-    // Add the path-history, if applicable (only applies to HL7 IGs)
-    if (/^https?:\/\/hl7.org\//.test(this.pkg.packageJSON.canonical)) {
-      this.ig.definition.parameter.push({
-        code: 'path-history',
-        value: `${this.pkg.packageJSON.canonical}/history.html`
-      });
+    // add non-omitted resources
+    this.config.resources?.forEach(resource => {
+      if (!resource.omit) {
+        this.ig.definition.resource.push(resource);
+      }
+    });
+    // check for some optional properties and add them if present
+    if (this.config.useContext?.length) {
+      this.ig.useContext = this.config.useContext;
     }
-
-    // Add the dependencies
-    if (this.pkg.packageJSON.dependencies) {
+    // add dependencies
+    if (this.config.dependencies?.length) {
+      this.ig.dependsOn = [];
       const igs = this.fhirDefs.allImplementationGuides();
-      for (const depId of Object.keys(this.pkg.packageJSON.dependencies)) {
-        if (depId === 'hl7.fhir.r4.core') {
+      for (const dependency of this.config.dependencies) {
+        if (dependency.packageId === 'hl7.fhir.r4.core') {
           continue;
         }
-        const depVersion = this.pkg.packageJSON.dependencies[depId];
-        // find the matching IG by id (for "current"/"dev" version) or id and version (for specific version)
-        const depIG = igs.find(
+        const dependencyIG = igs.find(
           ig =>
-            ig.packageId === depId &&
-            (ig.version === depVersion || 'current' === depVersion || 'dev' === depVersion)
+            ig.packageId === dependency.packageId &&
+            (ig.version === dependency.version ||
+              'current' === dependency.version ||
+              'dev' === dependency.version)
         );
-        if (depIG && depIG.url) {
+        if (dependencyIG?.url) {
           this.ig.dependsOn.push({
-            uri: `${depIG.url}`,
-            packageId: depId,
-            version: depVersion
+            uri: dependencyIG.url,
+            packageId: dependency.packageId,
+            version: dependency.version
           });
         } else {
           logger.error(
-            `Failed to add ${depId}:${depVersion} to ImplementationGuide instance.  Could not determine its canonical URL from the FHIR cache.`
+            `Failed to add ${dependency.packageId}:${dependency.version} to ImplementationGuide instance.  Could not determine its canonical URL from the FHIR cache.`
           );
         }
       }
-      if (this.ig.dependsOn.length === 0) {
-        delete this.ig.dependsOn;
-      }
+    }
+    if (this.config.global?.length) {
+      this.ig.global = this.config.global;
+    }
+
+    // Add the path-history, if applicable (only applies to HL7 IGs)
+    if (/^https?:\/\/hl7.org\//.test(this.config.url)) {
+      this.ig.definition.parameter.push({
+        code: 'path-history',
+        value: `${this.config.url}/history.html`
+      });
```

Comment:
I believe the use of `config.url` here should now be replaced with `config.canonical`.